desktop_version/src/Render.cpp

```diff
@@ -2040,6 +2040,143 @@ static void draw_roomname_menu(void)
+                {
+                    for (int y = 0; y < mapzoom; y++)
+                    {
+                        graphics.drawimage(2, mapxoff + 40 + (x * 12) + (i * (12 * mapzoom)), mapyoff + 21 + (y * 9) + (j * (9 * mapzoom)), false);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void rendermaplegend(void)
+{
+    // Draw the map legend, aka teleports/targets/trinkets
+
+    int zoom_offset_x;
+    int zoom_offset_y;
+    int zoom_mult = map.custommode ? map.customzoom : 1;
+
+    switch (zoom_mult)
+    {
+        case 4:
+            zoom_offset_x = 60;
+            zoom_offset_y = 35;
+            break;
+        case 2:
+            zoom_offset_x = 48;
+            zoom_offset_y = 26;
+            break;
+        default:
+            zoom_offset_x = 43;
+            zoom_offset_y = 22;
+            break;
+    }
+
+    int tile_offset = graphics.flipmode ? 3 : 0;
+
+    for (size_t i = 0; i < map.teleporters.size(); i++)
+    {
+        if (map.showteleporters && map.isexplored(map.teleporters[i].x, map.teleporters[i].y))
+        {
+            graphics.drawtile(zoom_offset_x + (map.teleporters[i].x * 12 * zoom_mult), zoom_offset_y + (map.teleporters[i].y * 9 * zoom_mult), 1127 + tile_offset);
+        }
+        else if (map.showtargets && !map.isexplored(map.teleporters[i].x, map.teleporters[i].y))
+        {
+            graphics.drawtile(zoom_offset_x + (map.teleporters[i].x * 12 * zoom_mult), zoom_offset_y + (map.teleporters[i].y * 9 * zoom_mult), 1126 + tile_offset);
+        }
+    }
+
+    if (map.showtrinkets)
+    {
+        for (size_t i = 0; i < map.shinytrinkets.size(); i++)
+        {
+            if (!obj.collect[i])
+            {
+                graphics.drawtile(zoom_offset_x + (map.shinytrinkets[i].x * 12 * zoom_mult), zoom_offset_y + (map.shinytrinkets[i].y * 9 * zoom_mult), 1086 + tile_offset);
+            }
+        }
+    }
+}
+
+void rendermapcursor(bool flashing)
+{
+    int mapwidth = map.custommode ? map.customwidth : 20;
+    int mapheight = map.custommode ? map.customheight : 20;
+    int mapzoom = map.custommode ? map.customzoom : 1;
+    int mapxoff = map.custommode ? map.custommmxoff : 0;
+    int mapyoff = map.custommode ? map.custommmyoff : 0;
+
+    if (!map.custommode && game.roomx == 109)
+    {
+        // Draw the tower specially
+        if (!flashing)
+        {
+            graphics.drawrect(40 + ((game.roomx - 100) * 12) + 2, 21 + 2, 12 - 4, 180 - 4, 16, 245 - (help.glow * 2), 245 - (help.glow * 2));
+        }
+        else if (map.cursorstate == 1)
+        {
+            if (int(map.cursordelay / 4) % 2 == 0)
+            {
+                graphics.drawrect(40 + ((game.roomx - 100) * 12), 21, 12, 180, 255, 255, 255);
+                graphics.drawrect(40 + ((game.roomx - 100) * 12) + 2, 21 + 2, 12 - 4, 180 - 4, 255, 255, 255);
+            }
+            if (map.cursordelay > 30) map.cursorstate = 2;
```

Comment:
I didn't realize the Tower code still mutated `map.cursorstate` in a delta function, this should be removed.
```suggestion
```